### PR TITLE
fix-circleci-golint

### PR DESCRIPTION
**Pull request by pieceofr**
+ fix-circle-ci error (mainly go-lint)
+ remove prefer IPV6 in p2p connection
+ pass bm-tester and make-blockchain

### CODE REVIEW

**Comment by lemonlatte** [comment]
ok to test

File: p2p/listener.go [hunk 1]
```diff
@@ -115,30 +112,31 @@ func (l *ListenHandler) handleStream(stream network.Stream) {
...
-					listenerSendError(rw, nodeChain, err, "-->Query Block: block not found", log)
-				}
-				respParams := [][]byte{result}
-				packed, err := PackP2PMessage(nodeChain, "B", respParams)
-				if err != nil {
-					listenerSendError(rw, nodeChain, err, "-->Query Block  Information", log)
-					break
+			if 1 == len(parameters) {
+				listenerSendError(rw, nodeChain, fault.MissingParameters, "-->Query Block: invalid parameter", log)
+				if 8 == len(parameters[0]) { //it 8 or 6 ??
```
Comment (lemonlatte) [inline p2p/listener.go]: This `if` statement should be in the same level to `1 == len(parameters)`.

File: p2p/listener.go [hunk 2]
```diff
@@ -176,26 +174,29 @@ func (l *ListenHandler) handleStream(stream network.Stream) {
...
+				randData, _ = PackRegisterData(nodeChain, fn, nType, reqID, reqMaAddrs, time.Now())
 				util.LogDebug(log, util.CoReset, fmt.Sprintf("Send back peer as a random node ID:%v addrs:%v", reqID.ShortString(), util.PrintMaAddrs(reqMaAddrs)))
 			} else { //Get a Random Node
-				randData, err = PackRegisterData(nodeChain, fn, nType, randPeerID, randListeners, randTs)
+				randData, _ = PackRegisterData(nodeChain, fn, nType, randPeerID, randListeners, randTs)
 				util.LogDebug(log, util.CoReset, fmt.Sprintf("Send a random node ID:%v addrs:%v", randPeerID.ShortString(), util.PrintMaAddrs(randListeners)))
 			}
-
+			if 0 == len(randData) { //deal with pack error here not above
+				listenerSendError(rw, nodeChain, fault.PackRandomNodeFail, "-><- Radom node", log)
```
Comment (lemonlatte) [inline p2p/listener.go]: `listenerSendError` sometimes `break` and sometimes not. Do you intend to do this for further logging?

File: p2p/listener.go [hunk 3]
```diff
@@ -176,26 +174,29 @@ func (l *ListenHandler) handleStream(stream network.Stream) {
 			randPeerID, randListeners, randTs, err := announce.GetRandom(reqID)
 			var randData [][]byte
 			if nil != err || util.IDEqual(reqID, randPeerID) { // No Random Node sendback this Node
-				randData, err = PackRegisterData(nodeChain, fn, nType, reqID, reqMaAddrs, time.Now())
```
[the remark this answers is not in the document]
Comment (pieceofr) [reply p2p/listener.go]: ok

File: p2p/listener.go (hunk 2 above)
Comment (pieceofr) [reply p2p/listener.go]: all should be break

File: p2p/listener.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (pieceofr) [reply p2p/listener.go]: ok

File: p2p/listener.go (hunk 1 above)
Comment (pieceofr) [reply p2p/listener.go]: b4, it use 1!=len(parameters) so it is in the same level. 
I try to use the exclusive way to write it 1==len(parameter) so it is in the same level.
but listenerSendError should not be here but in the buttom

Comment (lemonlatte) [reply p2p/listener.go]: OK. Understand

File: p2p/pack.go [hunk 4]
```diff
@@ -66,40 +66,45 @@ func UnPackRegisterData(parameters [][]byte) (nodeType string, id peerlib.ID, ad
 }
 
 //PackRegisterData pack node message into p2pMessage
-func PackRegisterData(chain, fn string, nodeType string, id peerlib.ID, addrs []ma.Multiaddr, ts time.Time) ([][]byte, error) {
+func PackRegisterData(chain, fn string, nodeType string, id peerlib.ID, addrs []ma.Multiaddr, ts time.Time) [][]byte {
 	typePacked := []byte(nodeType)
 	idPacked, err := id.Marshal()
```
Comment (lemonlatte) [inline p2p/pack.go]: Since there will not be an error in this function, we can use `idPacked, _ := id.Marshal()` without checking its error.

Comment (pieceofr) [reply p2p/pack.go]: I have a second thought about this. When someone reads the code, would this person curious about why?  Would leave the error here cause less confusion?

**Pull request merged**